pysat/tests/test_orbits.py

```diff
@@ -403,183 +423,76 @@ def filter_data(inst):
-
-    def test_repeated_orbit_calls_symmetric_multi_day_off_0_UT(self):
-        """ Test successful orbit calls for days about a time off 00:00 UT
-        """
-        self.stime -= dt.timedelta(days=1)
-        self.testInst.load(date=self.stime)
-        self.testInst.orbits.next()
-        control = self.testInst.copy()
-        for j in range(20):
-            self.testInst.orbits.next()
-        for j in range(20):
-            self.testInst.orbits.prev()
-        assert all(control.data == self.testInst.data)
-
-    def test_repeated_orbit_calls_antisymmetric_multi_day_off_0_UT(self):
-        """ Test successful orbit calls for different days about a time off 0 UT
-        """
-        self.stime -= dt.timedelta(days=1)
-        self.testInst.load(date=self.stime)
-        self.testInst.orbits.next()
-        control = self.testInst.copy()
-        for j in range(10):
-            self.testInst.orbits.next()
-        for j in range(20):
-            self.testInst.orbits.prev()
-        for j in range(10):
-            self.testInst.orbits.next()
-        assert all(control.data == self.testInst.data)
-
-    def test_repeated_orbit_calls_antisymmetric_multi_multi_day_off_0_UT(self):
-        """ Test successful orbit calls for more days about a time off 0 UT
-        """
-        self.stime -= dt.timedelta(days=1)
-        self.testInst.load(date=self.stime)
-        self.testInst.orbits.next()
-        control = self.testInst.copy()
-        for j in range(20):
-            self.testInst.orbits.next()
-        for j in range(40):
-            self.testInst.orbits.prev()
-        for j in range(20):
-            self.testInst.orbits.next()
-        assert all(control.data == self.testInst.data)
-
-    def test_repeated_orbit_calls_antisymmetric_multi_day_0_UT(self):
-        self.testInst.load(date=self.stime)
-        self.testInst.orbits.next()
-        control = self.testInst.copy()
-        for j in range(10):
-            self.testInst.orbits.next()
-        for j in range(20):
-            self.testInst.orbits.prev()
-        for j in range(10):
-            self.testInst.orbits.next()
-        assert all(control.data == self.testInst.data)
-
-    def test_repeated_orbit_calls_antisymmetric_multi_multi_day_0_UT(self):
+    @pytest.mark.parametrize("iterations", [(10), (20)])
+    def test_repeated_orbit_calls(self, iterations):
+        """Test that repeated orbit calls are reversible."""
         self.testInst.load(date=self.stime)
         self.testInst.orbits.next()
         control = self.testInst.copy()
-        for j in range(20):
+        n_time = []
+        p_time = []
+        for j in range(iterations):
+            n_time.append(self.testInst.index[0])
             self.testInst.orbits.next()
-        for j in range(40):
+        for j in range(iterations):
             self.testInst.orbits.prev()
-        for j in range(20):
-            self.testInst.orbits.next()
+            p_time.append(self.testInst.index[0])
```

Comment:
This "reversability" code snippit is repeated a lot.  It could probably be an evaluation method in the core class.